### PR TITLE
include_patterns

### PR DESCRIPTION
**Pull request by jasonfellows**
- Adds include_patterns feature to route_downcaser

### CODE REVIEW

File: lib/route_downcaser/downcase_route_middleware.rb [hunk 1]
```diff
@@ -15,11 +15,18 @@ def _call(env)
...
       }
 
-      # Don't touch anything, if uri/path is part of exclude_patterns
-      if exclude_patterns_match?(env['REQUEST_URI']) or exclude_patterns_match?(env['PATH_INFO'])
+      # Don't touch anything if uri/path is part of exclude_patterns
+      if exclude_patterns_match?(env['REQUEST_URI']) || exclude_patterns_match?(env['PATH_INFO'])
         return @app.call(env)
       end
 
+      # Don't touch anything uf there are include_patterns and uri/path isn't included
```
Comment (valakirka) [inline lib/route_downcaser/downcase_route_middleware.rb]: Minor: `uf` -> `if`

File: test/route_downcaser_test.rb [hunk 2]
```diff
@@ -76,25 +77,83 @@ class ExcludePatternsTests < ActiveSupport::TestCase
     end
   end
 
+  class IncludePatternsTests < ActiveSupport::TestCase
+    setup do
+      @app = MyMockApp.new
```
Comment (sareg0) [inline test/route_downcaser_test.rb]: what is 'MyMockApp'? Can you explain to me.

File: test/route_downcaser_test.rb [hunk 3]
```diff
@@ -103,36 +162,75 @@ class RedirectTrueTests < ActiveSupport::TestCase
...
+      RouteDowncaser.configuration do |config|
+        config.redirect = true
+        config.exclude_patterns = nil
+        config.include_patterns = [/hello\//i]
+      end
+    end
+
+    test "when redirect is true, it does not redirect if REQUEST_URI doesn't match include patterns" do
```
Comment (sareg0) [inline test/route_downcaser_test.rb]: just from the perspective of someone attempting to understand the code, by reading the tests, it isn't very clear when read in plain english. I have some suggestions

`when redirect is true, it does not redirect if REQUEST_URI doesn't match include_patterns`
(this would indicate there is a method that the reader can go and look at to understand what 'include_patterns' is).

`when redirect is true, it does not redirect if REQUEST_URI does not match an included pattern`
(just reads better in plain english)

Also, from line 181, perhaps an improvement would be
`when redirect is true, it does not redirect if PATH_INFO matches an excluded pattern`
or
`when redirect is true, it does not redirect if PATH_INFO matches exclude_patterns`

While these comments are minor, such improvements helps speed up my understanding of the code, because the explanation of the test just makes total sense when read as a sentence :)

**Comment by sareg0** [comment]
I made a minor suggestion, but all in all the code LGTM.

**Comment by valakirka** [comment]
LGTM

**Comment by juliabecker** [comment]
LGTM

**Comment by stefanozanella** [comment]
LGTM

**Comment by stefanozanella** [comment]
@jasonfellows we didn't open a PR in the original project in the end, did we?

**Comment by jasonfellows** [comment]
@stefanozanella Not yet. I was waiting to use it in production for a bit.
